### PR TITLE
Adds dim kwarg support for cross entropy loss

### PR DESCRIPTION
**Pull request by yashika-git**
I've tried adding the required `dim` support.
Is this how it is to be done?
If yes, I'll update the inline comments as well.
Thank You

Fixes #77256

cc @albanD @mruberry @jbschlosser @walterddr @saketh-are

### CODE REVIEW

**Comment by yashika-git** [comment]
@jbschlosser, Could you please review it?
Thank You

**Comment by kit1980** [comment]
@jbschlosser Could you take a look?

**Comment by malfet** [comment]
It would be good to document a new argument to embedded docstring. 
But also, why is this needed to be part of this operation?

**Comment by yashika-git** [comment]
@malfet, I have documented the new argument to embedded docstring.
For the need of argument, [this issue]( https://github.com/pytorch/pytorch/issues/77256#issue-1232810168) can be referred.

**Review by malfet — APPROVED** [review]
This looks consistent with original request in the issue.
@jbschlosser , @janeyx99 does the change look good to you?

File: torch/nn/functional.py [hunk 1]
```diff
@@ -2986,6 +2987,8 @@ def cross_entropy(
             of smoothing when computing the loss, where 0.0 means no smoothing. The targets
             become a mixture of the original ground truth and a uniform distribution as described in
             `Rethinking the Inception Architecture for Computer Vision <https://arxiv.org/abs/1512.00567>`__. Default: :math:`0.0`.
+        dim (int, optional): Specifies the dimension of the classes (the dimension across which softmax is computed).
+            Default: 1
```
Comment (albanD) [inline torch/nn/functional.py:2991]: The default here is actually `None` ?

Comment (yashika-git) [reply torch/nn/functional.py:2991]: @albanD, I set Default as 1 because I thought if no value is specified by the user for dim, then by default softmax would be computed across the first dimension.

**Comment by yashika-git** [comment]
@jbschlosser, could you please review this PR? 
Thanks

File: torch/nn/functional.py (hunk 1 above)
Comment (albanD) [reply torch/nn/functional.py:2991]: Yeah I think that is fine but we make these doc match 1-1 the implementation. You can say: the values default to `1` if not provided. Default: None.

Comment (yashika-git) [reply torch/nn/functional.py]: @albanD, Got it, made Default to ``None``

**Review by janeyx99 — APPROVED** [review]

**Review by janeyx99 — CHANGES_REQUESTED** [review]
This looks good! I canceled the merge because I think it would be best to add a test case in test_nn.py.

**Comment by janeyx99** [comment]
@yashika-git hello! I wanted to follow up and see if you were able to add a test case to ensure the behavior is as expected. A good example to derive inspiration is `test_cross_entropy_loss` in `test/test_nn.py`

File: torch/nn/functional.py [hunk 2]
```diff
@@ -3016,6 +3020,8 @@ def cross_entropy(
         >>> loss = F.cross_entropy(input, target)
         >>> loss.backward()
     """
+    if dim is not None:
+        input = input.swapaxes(dim, 1)
```
Comment (jbschlosser) [inline torch/nn/functional.py:3024]: The contract we provide for `__torch_function__` support is that the full Python API should be overrideable.

Given this, the `handle_torch_function()` logic should be first (before the lines added here), and `dim=dim` should be passed to the `handle_torch_function()` call.

File: torch/nn/functional.py [hunk 3]
```diff
@@ -2950,6 +2950,7 @@ def cross_entropy(
     reduce: Optional[bool] = None,
     reduction: str = "mean",
     label_smoothing: float = 0.0,
+    dim: Optional[int] = None,
```
Comment (jbschlosser) [inline torch/nn/functional.py:2953]: I think we also want this arg for module form `nn.CrossEntropyLoss`, no?

**Review by jbschlosser — CHANGES_REQUESTED** [review]
Thanks for the contribution! As @janeyx99 mentioned, we definitely need tests for the additions (`test/test_nn.py` is the usual dumping grounds), and I think we'll want similar changes made to the module form of cross entropy as well.

**Comment by yashika-git** [comment]
> @yashika-git hello! I wanted to follow up and see if you were able to add a test case to ensure the behavior is as expected. A good example to derive inspiration is `test_cross_entropy_loss` in `test/test_nn.py`

Hello @janeyx99, Thanks for sharing the test example. I was a little occupied the past week so couldn't add the test case, will try adding it this week.

File: torch/nn/functional.py (hunk 3 above)
Comment (yashika-git) [reply torch/nn/functional.py:2953]: @jbschlosser, sure, I'll try adding the arg in ``nn.CrossEntropyLoss`` as well.

File: torch/nn/functional.py (hunk 2 above)
Comment (yashika-git) [reply torch/nn/functional.py:3024]: @jbschlosser, Alright, I'll modify this.
